src/core/registration.c

```diff
@@ -249,6 +252,10 @@ MsQuicRegistrationShutdown(
 
         CxPlatDispatchLockAcquire(&Registration->ConnectionLock);
```

Comment:
Maybe only do this if `ShuttingDown` isn't already `TRUE`?

Reply:
Probably. I switched it. Should I dbg assert if its already true?

Reply:
No. Generally, we allow shutdown functions to be called multiple times, with them being no-op additional calls.